mailbox/grpc_noise_conn.go

```diff
@@ -215,6 +215,17 @@ func (c *NoiseGrpcConn) ClientHandshake(_ context.Context, _ string,
 		return nil, nil, err
 	}
 
+	// If the server has indicated that there will be authData sent in
```

Comment:
Can't this just be gated based on the protocol version? So v0 is what we have rn, and then in v1 we expect a payload. The payload can just be a regular message though, and we wouldn't need to do anything special here other than do a normal call to `ReadMessage` here, as for v1 we _always_ send that extra message. 

FWIW, from a network perspective, it's more or less the same as having dynamic data be sent along with the payload in the earlier act. Apologies that I fell behind on this PR a bit, but my initial comment was mainly meant to force us to ackledge the implications of dynamic payloads.

Reply:
cool. yeah the only reason i did it like this (ie, not using `ReadMessage`) is because of the current 2 byte length prefix limit for normal messages. But if it is fine for me to change that, ie use 2 byte length prefix for v0 comms and use 4 byte length prefix for all v1 comms, then yes that makes things way simpler